src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -675,24 +675,37 @@ void DistributedSearchTreeImpl<DeviceType>::filterResults(
       return lhs.second > rhs.second;
     }
   };
+
+  int const n_results = lastElement(offset);
+  Kokkos::View<PairIndexDistance *, DeviceType> buffer(
+      Kokkos::ViewAllocateWithoutInitializing("buffer"), n_results);
```

Comment:
This seems like an overkill in terms of size, as you double the storage. You could make the calculation of `new_offset` above a `parallel_reduce` with simultaneous computing of the max size.

Reply:
I am fixing a bug, not prematurely optimizing

Reply:
Only partially true. The previous (buggy) implementation had a limited (256) memory footprint. The new one is unlimited and is dependent on the number of found results for all queries, which potentially is expensive. I consider this a regression.

Reply:
I would be fine with merging as is and looking into the performance impact after that.